tutorials/basic-topics/sorting/insertion-sort.md

```diff
@@ -11,4 +11,204 @@ keywords:
+            height_of_students[j + 1] = height_of_students[j];
+            j--;
+        }
+        height_of_students[j + 1] = nums;
+    }
+}
+
+int main()
+{
+    int height_of_students[] = {175, 180, 171, 160, 168};
+    int size_of_arary = sizeof(height_of_students) / sizeof(height_of_students[0]);
+
+    insertionSort(height_of_students, size_of_arary);
+
+    for (auto it : height_of_students) {
+        cout << it << " ";
+    }
+}
+```
+
+### Java Implementation
+
+```java
+import java.util.Arrays;
+class InsertionSort {
+	void insertion_sort(int height_of_students[],int size_of_arary) {
+		for (int i = 1; i < size_of_arary; ++i) {
+			int nums = height_of_students[i];
+			int j = i - 1;
+
+			while (j >= 0 && height_of_students[j] > nums) {
+				height_of_students[j + 1] = height_of_students[j];
+				j = j - 1;
+			}
+			height_of_students[j + 1] = nums;
+		}
+	}
+
+	public static void main(String args[]) {
+		int height_of_students[] = {175, 180, 171, 160, 168};
+        int size_of_arary = height_of_students.length;
+
+		InsertionSort obj = new InsertionSort();
+		obj.insertion_sort(height_of_students,size_of_arary);
+
+		for(int element : height_of_students){
+			System.out.print(element + " ");
+		}
+	}
+}
+```
+
+### Python Implementation
+
+```python
+def insertionSort(height_of_students,size_of_array):
+
+    for i in range(1, size_of_array):
+        nums = height_of_students[i]
+        j = i - 1
+              
+        while j >= 0 and height_of_students[j] > nums:
+            height_of_students[j + 1] = height_of_students[j]
+            j = j - 1
+        
+        height_of_students[j + 1] = nums
+
+
+height_of_students = [175, 180, 171, 160, 168]
+size_of_array = len(height_of_students)
+insertionSort(height_of_students,size_of_array)
+print(height_of_students)
+```
+
+## Time Complexity
+
+### Worst Case Complexity
+
+The worst case would be when we have to compare `nums` with every element in the left of it that would we `(n-1)` comparisons for `n` elements. Means the array would be in descending order.
+
+So, the complexity is = $\mathcal{O}(n^2)$
+
+### Best Case Complexity
+
+The best case would be when the comparisons with `nums` be zero. Means the array already is in sorted order. So, Only outer loop over array would run.
+
+So, the complexity is = $\mathcal{O}(n)$
+
+### Average Case Complexity
+
+The average case would be when the array is neither in ascending order nor in descending order.
+
+So, the complexity is = $\mathcal{O}(n^2)$
+
+## Space Complexity
+
+We are not using any extra space apart from the array we have to sort. So, the space complexity is = $\mathcal{O}(1)$
+
+export const suggestedProblems = [
+  {
+    "problemName": "1365 - How Many Numbers Are Smaller Than the Current Number",
+    "difficulty": "Easy",
+    "leetCodeLink": "https://leetcode.com/problems/how-many-numbers-are-smaller-than-the-current-number/",
+    "solutionLink": "https://leetcode.com/submissions/detail/812551570/"
```

Comment:
Don't link a submission. Link to a page in `solutions`. If not exist, either create it or leave it blank.